### PR TITLE
pixart: Add support for more devices

### PR DESCRIPTION
**Pull request by hughsie**
Type of pull request:
- [ ] New plugin (Please include [new plugin checklist](https://github.com/fwupd/fwupd/wiki/New-plugin-checklist))
- [ ] Code fix
- [X] Feature
- [ ] Documentation

### CODE REVIEW

**Comment by hughsie** [comment]
@jyu99 I've rebased the changes and fixed up a lot of issues. Lets carry on discussion on this PR.

File: plugins/pixart-rf/fu-pxi-device.c [hunk 1]
```diff
@@ -599,10 +615,25 @@ static gboolean
...
+	if (!fu_pxi_device_get_raw_info (self, &hid_raw_info ,error))
+		return FALSE;
 	if (!fu_pxi_device_fw_ota_init (self, error))
 		return FALSE;
 	if (!fu_pxi_device_fw_get_info (self, error))
 		return FALSE;
+
+	/* extra GUID */
+	g_string_ascii_up (ble_name);
+	fu_common_string_replace (ble_name, " ", "_");
```
Comment (superm1) [inline plugins/pixart-rf/fu-pxi-device.c:631]: is there any other filtering that might be needed for this string?

**Review by superm1 — APPROVED** [review]
Just one nit from me, else looks fine.

File: plugins/pixart-rf/fu-pxi-device.c (hunk 1 above)
Comment (hughsie) [reply plugins/pixart-rf/fu-pxi-device.c:631]: Do we have any ideas for this? Quite a few plugins do this now. I think the baseline ought to be ASCII, no spaces, uppercase.

Comment (superm1) [reply plugins/pixart-rf/fu-pxi-device.c:621]: yeah just with the all the fuzzing stuff you've done lately in mind, it's definitely "untrusted" data.  I think that's spot on, ASCII only, no spaces.  uppercase we need to double check all the other plugins that they're not using lowercase.

Comment (hughsie) [reply plugins/pixart-rf/fu-pxi-device.c:621]: > we need to double check all the other plugins

At least one is. We can't change the instance IDs now, the GUIDs are essentially _API_.

Comment (superm1) [reply plugins/pixart-rf/fu-pxi-device.c:621]: then just got with uppercase for everything new I guess

**Comment by jyu99** [comment]
If we have more than one device to update at same time, the fwupd is execute update task concurrent by multi-thread, or just execute the task sequentially ?

**Comment by hughsie** [comment]
> fwupd execute updating task concurrent

No, all of fwupd is (currently) single threaded. All updates are always done sequentially.

**Comment by jyu99** [comment]
Hi  I have a problem about GUID

**Comment by superm1** [comment]
>Hi I have a problem about GUID

What problem?

**Pull request merged**